helm/resource_release.go

```diff
@@ -164,20 +178,54 @@ func resourceReleaseCreate(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return err
 	}
+	name := d.Get("name").(string)
+out:
+	for {
```

Comment:
maybe you can move this logic to a function

Reply:
yes. Double check if the logic makes sense. Especially if the state of the chart is unknown, should we abort, wait or try to install the new chart anyway?